src/inputs/dnstap/DnstapInputStream.cpp

```diff
@@ -56,15 +60,14 @@ void DnstapInputStream::_read_frame_stream_file()
                 _logger->warn("dnstap data is wrong type or has no message, skipping frame of size {}", len_data);
                 continue;
             }
-            if (_filter_host) {
-                if (d.message().has_query_address()) {
-                    if (!_match_subnet(d.message().query_address())) {
+            if (_f_enabled[Filters::OnlyHosts]) {
+                if (d.message().has_query_address() && d.message().has_response_address()) {
+                    if (!_match_subnet(d.message().query_address()) || !_match_subnet(d.message().response_address())) {
                         continue;
                     }
-                    if (d.message().has_response_address()) {
-                        if (!_match_subnet(d.message().response_address())) {
-                            continue;
-                        }
+                } else if (d.message().has_query_address()) {
```

Comment:
this (and below) can collapse to one if now, makes it a bit easier to read I think

Reply:
collapse in a `||` comparison? because I should know if it exists to perform match;

Reply:
i just meant 
```
} else if (d.message().has_query_address() && !_match_subnet(d.message().query_address())) {
    continue;
}
```

Reply:
ok

Reply:
done